controllers/history.controller.js

```diff
@@ -0,0 +1,88 @@
+const History = require("../models/history-model");
+const User = require("../models/user-model");
+const { concat,remove }=require("lodash");
+
+const getHistory=async(req,res)=>{
+  const history=await History.find({});
+  res.json({success:true,history});
+}
+
+const findUserHistory = async (req, res, next, userId) => {
+  try {
+    let user = await User.findOne({ _id: userId });
+    if (!user) {
+      res.status(404).json({
+        success: false,
+        message: "Invalid user! Kindly register to continue",
+      });
+      throw Error("Invalid User");
+    }
+    let history = await History.findOne({ userId });
+
+    if (!history) {
+      history = new History({ userId, videos: [] });
+      history = await history.save();
+    }
+    req.history = history;
+    next();
+  } catch (error) {
+    res.status(500).json({
+      success: false,
+      message: "Unable to retrive user's watch history",
+      errorMessage: error.message,
+    });
```

Comment:
It's a good practice to console the errors !

Reply:
Yes